lib/paseto.d.ts

```diff
@@ -0,0 +1,205 @@
+declare namespace Paseto {
+    interface IPasetoKey<P extends IProtocol> {
+        /**
+         * complete construction asynchronously
+         */
+        inject(rkey: Buffer): Promise<void>;
```

Comment:
Is there any reason (here and below) why you didn't include the callback function in both the signature definition nor the corresponding callback signature in the return definition? I consider callbacks to be a perfectly valid use of the library and would like to see them represented, if possible.

Reply:
🤔 I don't know any good way to do it...

One way is to duplicate everything :

```Typescript
interface IProtocol {
    /**
     * generate a private key for use with the protocol
     */
    private(): Promise<PrivateKey<this>>;
    private<T>(cb: (err: Error, key: PrivateKey<this>) => T): T

    /**
     * generate a symmetric key for use with the protocol
     */
    symmetric(): Promise<SymmetricKey<this>>;
    symmetric<T>(cb: (err: Error, key: SymmetricKey<this>) => T): T
```

@sloonz May have a better idea...

Reply:
Why the generic `<T>` ? I think it should be something like : 

```
symmetric(cb: (err: Error, key: SymmetricKey<this>) => any): void;
```

The rationale behind not exporting the callback API was that a typescript user will most likely use await/async anyway.

I don’t think there’s a clean way to do this without duplicating the signatures. Well, I see a way, but not clean at all : 

```typescript
type PromiseOrCallbackMethod<T1,R> = {
    (arg: T1: Promise<R>;
    (arg: T1, cb: (err: Error, result: R) => any): void;
};

interface PrivateKey {
    hex: PromiseOrCallbackMethod<string, void>;
}
```

Reply:
I used T type because the function `ret` used to return a promise is implemented as :
``` Javascript
return callback.apply(this, args);
```
(I think it should be)
``` Javascript
return void callback.apply(this, args);
```

Reply:
What would be the harm in duplicating the signatures beyond just the literal duplication? If there is none, then I'm fine with that.

> The rationale behind not exporting the callback API was that a typescript user will most likely use await/async anyway.

This is a fair point, and I think is a good argument for not making the signatures unnecessarily complex if duplicating them is unreasonable.

Reply:
done